src/metrics/listener.ts

```diff
@@ -135,17 +135,24 @@ export class MetricsListener {
     forceAsync: boolean,
     ...tags: string[]
   ) {
+    const dist = new Distribution(name, [{ timestamp: metricTime, value }], ...tags);
+
     if (this.isExtensionRunning) {
-      this.statsDClient?.distribution(name, value, undefined, tags);
-      return;
+      const isMetricTimeValid = Date.parse(metricTime.toString()) > 0;
+      if (isMetricTimeValid) {
+        // Only create the processor to submit metrics to the API when a user provides a valid timestamp as
+        // Dogstatsd does not support timestamps for distributions.
+        this.currentProcessor = this.createProcessor(this.config, this.apiKey);
+      } else {
```

Comment:
Let's try to avoid having `if-else` blocks. Also, wouldn't this need a return? This seems like it would process the metric if the extension is running but if `this.config.logForwarding` is set to `true`, it will both process it and then write it to `stdout`.

Reply:
The `if` here just creates the processor since it's typically never needed when using the extension, except for this special case, otherwise always send to statsd. Later on the metric is added to the processor at the end of the function (unchanged). So if `logForwarding` is true and the extension is running, then it will return on L153 and not process

Reply:
You are right, thanks for explaining!